stmtdiagnostics: remove unused JSON trace datum

yuzefovich opened the pull request:
In 2d05a54 we stopped writing JSON
trace datums to the system table, but some remnants of that work still
remained. This commit cleans that up.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/65583)
<!-- Reviewable:end -->

RaduBerinde reviewed (changes requested):
Haha, noooooo! You are removing the traces from the bundle, which are super useful. 2d05a54 stopped writing it to the system table, not the bundle.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/65583#-:-MaGPQoeDSy3-GXkm0dt:bideofi)* status: :shipit: complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

yuzefovich reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/65583)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @yuzefovich)

___
*[pkg/sql/explain_bundle.go, line 167 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/65583#-MaGRBCE-17DnEjklZ-O:-MaGRBCE-17DnEjklZ-P:b3td412) ([raw file](https://github.com/cockroachdb/cockroach/blob/040a7af95f2dc62e84498a3a90b73ead8ad7c8dd/pkg/sql/explain_bundle.go#L167)):*
> ```Go
> 	b.addDistSQLDiagrams()
> 	b.addExplainVec()
> 	traceJSON := b.addTrace()
> ```

I see - I need to remove unused local variable `traceJSON` - my bad, will update shortly.


<!-- Sent from Reviewable.io -->

yuzefovich committed:
stmtdiagnostics: remove unused JSON trace datum

In 2d05a5421551ad8e2cb8fa7fd6a8c08b62e26a13 we stopped writing JSON
trace datums to the system table, but some remnants of that work still
remained. This commit cleans that up.

Release note: None

yuzefovich reviewed (commented):
Updated, PTAL.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/65583#-:-MaGTG1N9sAnDR0p-wz8:b-rr168b)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @yuzefovich)



<!-- Sent from Reviewable.io -->

RaduBerinde reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 2 files at r1, 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/65583#-:-MaGe6Yl6Rix6p-nLk8n:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @yuzefovich)



<!-- Sent from Reviewable.io -->

yuzefovich commented:
TFTR!

bors r+
